### PR TITLE
Serverless support: define a protobuf for the plugins that implement external storage of identities

### LINKED ISSUE #2067: Serverless support: define a protobuf for the plugins that implement external storage of identities
Define a protobuf that will be the contract used by the plugins that implement the capability to externally store issued identities.

### PR DESCRIPTION
**Pull request by MarcosDY**
Add SVID Store proto
It allow creating a new type of plugins that will be used to store X509-SVIDs on different platform where attestation is not possible

**Which issue this PR fixes**
fixes #2067

### CODE REVIEW

File: proto/spire/agent/svidstore/svidstore.proto
Comment (rturner3) [inline proto/spire/agent/svidstore/svidstore.proto:35]: Can we use the [SPIFFEID](https://github.com/spiffe/spire/blob/49678bbfb34db42722075543d0ee3820a1fb5d3a/proto/spire/types/spiffeid.proto#L7) message type here rather than a `string`?

File: proto/spire/agent/svidstore/svidstore.proto
Comment (rturner3) [inline proto/spire/agent/svidstore/svidstore.proto:38]: This should be `repeated bytes` since it represents one or more X.509 certificates.

File: proto/spire/agent/svidstore/svidstore.proto
Comment (rturner3) [inline proto/spire/agent/svidstore/svidstore.proto:44]: Similar comment here, this should be `repeated bytes`.

File: proto/spire/agent/svidstore/svidstore.proto
Comment (MarcosDY) [inline proto/spire/agent/svidstore/svidstore.proto:24]: I use common selectors because our cache maintain selectors using it, `but` may we use `spire.types.Selector` (https://github.com/spiffe/spire/blob/master/proto/spire/types/selector.proto#L5) instead?
At least in this proto and in a future we may update our cache to use new protos, with that we'll void parsing protos we get from SPIRE Server API.
@azdagron @evan2645 @amartinezfayo @rturner3  any ideas about it?

Comment (rturner3) [reply proto/spire/agent/svidstore/svidstore.proto:25]: If we want to use `spire.types.Selector` rather than `spire.common.Selectors` in the Agent cache, we'd probably have to either:

1. Make a breaking change to the `WorkloadAttestor` Agent plugin interface to use the type from `spire.types` and update all the Agent code to use the `spire.types` version

or

2. Translate instances of `spire.common.Selector` in Agent code to `spire.types.Selector` in the WorkloadAttestor handler and use `spire.types.Selector` everywhere else in the Agent code

Either option seems feasible to me, although with option 1, we would have to wait till a major release to break the `WorkloadAttestor` contract.

Comment (MarcosDY) [reply proto/spire/agent/svidstore/svidstore.proto:25]: mmm, make sense, I'm just worried about using `common` selector in a new proto,
I dont know if it makes sense to use `types` selectors here, we can create them from common selectors before calling `PutX509SVID`,
but we can update it after we update cache to use new `types` in a future of course.

Comment (azdagron) [reply proto/spire/agent/svidstore/svidstore.proto:25]: Just a drive by comment here :) The `types` package was introduced for the APIs. In the past there has not been a clear line between API related messages and plugin related messages. Coupling them together, while attractive from a DRY perspective, isn't really attractive from a "these things should be able to grow independently" perspective.

Considering all the other plugins still rely on the common package, I'd be hesitant to go one-off just for this plugin. IMO, we should continue to use the common package for plugins. At some point in time we can make a change to types if we want to, but now doesn't seem to be the right time.

**Review by rturner3 — APPROVED** [review]
LGTM, thanks @MarcosDY !

File: proto/spire/agent/svidstore/svidstore.proto
Comment (rturner3) [inline proto/spire/agent/svidstore/svidstore.proto:36]: I chatted with the other maintainers about this, and we ultimately decided we wanted to keep the types used in the SPIRE core RPCs separate from the types used in plugin protocol buffers for maintainability reasons. There are times when we might want to augment API types that may affect the plugin RPCs in undesirable ways if they share the same types, and vice versa. I think after we change this back to a `string`, I am good to merge this.

I apologize for the back and forth on this one since I originally led you down this road. We definitely have some room to improve in protocol buffer management and documentation in the project to make it more clear for contributors to know which types should be used in which RPCs.

**Review by azdagron — APPROVED** [review]
LGTM! Thanks, @MarcosDY .

**Pull request merged**
